lib/sdk/storage.js

```diff
@@ -52,6 +53,17 @@ async function getAsset(assetReference, directory, name, disableDownload) {
         const source = new Asset(assetReference, directory, name);
         if (disableDownload) {
             console.log(`Skipping source file download for ${AssetComputeLogUtils.redactUrl(source.url)}`);
+            const protocol = URL.parse(assetReference.url).protocol;
+            if (protocol === DATA_PROTOCOL) {
+                await datauri.download(assetReference, source.path);
+                if(!fileExistsAndIsNotEmpty(source.path)){
+                    throw new SourceUnsupportedError(`Invalid or missing local file ${source.path}`);
+                }
+                const preSignedUrl = await datauri.getPreSignedUrl(source.path);
```

Comment:
Consider adding logging, so we can easily identify where something failed if it fails.

Reply:
added

Reply:
done